### PR TITLE
Remove function prototypes from backend pt6

### PR DESCRIPTION
**Pull request by dkorpel**

### CODE REVIEW

File: compiler/src/dmd/backend/symbol.d [hunk 1]
```diff
@@ -327,7 +327,7 @@ version (SCPP_HTOD)
 
 @trusted
 extern (C)
-Symbol * symbol_calloc(const(char)[] id)
+Symbol * symbol_calloc(const(char)[] id) @nogc
```
Comment (thewilsonator) [inline compiler/src/dmd/backend/symbol.d:330]: unrelated? also why have a trailing `@nogc` with a leading `@trusted`?

**Review by thewilsonator — APPROVED** [review]
This otherwise looks good.

File: compiler/src/dmd/backend/symbol.d (hunk 1 above)
Comment (dkorpel) [reply compiler/src/dmd/backend/symbol.d:330]: The prototype was annotated `@nogc`, but the implementation not, so importing it without this annotation gives an error.

Comment (thewilsonator) [reply compiler/src/dmd/backend/symbol.d:330]: Ah, OK. would be nice to be consistent with the leading/trailing of attributes though.

Comment (dkorpel) [reply compiler/src/dmd/backend/symbol.d]: Done

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**